### ISSUE TITLE
Protobuf.js 6.8 makes version-incompatible changes to the message object prototype

### ISSUE THREAD

**Issue by murgatroid99** [dcodeIO/protobuf.js#832]
protobuf.js version: 6.8

Originally reported at grpc/grpc#11484.

Expected: `MessageClass.decode(buffer).toObject(options)` returns a plain object that is structurally equivalent to the decoded message.

Actual: `MessageClass.decode(buffer).toObject` is not a function.

This is a change from Protobuf.js 6.7.

**Comment by dcodeIO** [comment]
Related. https://github.com/dcodeIO/protobuf.js/issues/828#issuecomment-307877338

**Comment by murgatroid99** [comment]
I understand. Personally, I think you are overestimating the expectations that people have for major version bumps, and underestimating the negative impact of not following semantic versioning, even with the mentioned documentation and warning.

**Comment by orbiteleven** [comment]
I'm not sure that whatever historical reasoning one has for not being semver compatible outweighs the benefits of being semver compatible moving forward. I'm guessing you'll get a lot of complaints about this.

**Comment by sbowler** [comment]
I'll ignore the versioning debate since I'm mostly just glad that I don't have to write this code myself, but I do think at the very least it would be nice to have a small note on the releases page stating breaking changes for releases like 6.8.0.

Some simple statements like
> `from` has been dropped in favor of `fromObject` and both this one and `toObject` are solely static methods now.

would be well worth saving you some time dealing with questions and issues from people because of it. All the same, thanks for working on this project.

**Comment by dcodeIO** [comment]
Yeah, while it's in the [CHANGELOG](https://github.com/dcodeIO/protobuf.js/blob/master/CHANGELOG.md). I missed to add more details to the release note.

**Comment by seishun** [comment]
This makes for a somewhat awkward API. If you have a message class `T` and a Buffer `buf` which you want to turn into a plain object, you now have to write `T.toObject(T.decode(buf))`, repeating `T` twice. This is something you'd want to avoid if `T` has a long name.

**Comment by dcodeIO** [comment]
Closing this issue because it has been inactive for more than two years and the project has changed substantially.

If this is still relevant, please open a fresh issue with current reproduction steps, version details, and any updated context.

Please also consider that maintaining protobuf.js takes time and effort. If the project is valuable to you or your organization, sponsorship is appreciated.

Thanks!
